### PR TITLE
Fix/multiple providers

### PR DESCRIPTION
**Pull request by Despire**
Closes https://github.com/berops/claudie/issues/1296 
Closes https://github.com/berops/claudie/issues/1312

This PR also introduces a limit on names for:

clusters: 28 chars.
nodepools: 14 chars.

Note: Mostly the changes are in the terraform files where everything has a "provider.specName" suffix so that no collisions will be generated when using more than one of the same provider

### CODE REVIEW

**Review by cloudziu — APPROVED** [review]
LGTM 👍

File: internal/manifest/manifest.go [hunk 1]
```diff
@@ -24,25 +24,25 @@ type Provider struct {
 }
 
 type HetznerDNS struct {
-	Name     string `validate:"required" yaml:"name"`
+	Name     string `validate:"required,max=15" yaml:"name"`
```
Comment (JKBGIT1) [inline internal/manifest/manifest.go:27]: Shouldn't we write in the [API reference](https://docs.claudie.io/latest/input-manifest/api-reference/#providers) that the name of the cloud provider is limited by 15 chars?

**Review by JKBGIT1 — APPROVED** [review]
LGMT :+1:
Just one more question. Why do we use region in the `provider.alias` for most of the cloud providers and some (Hetzner) we don't?

**Comment by Despire** [comment]
> Why do we use region in the provider.alias for most of the cloud providers and some (Hetzner) we don't?

This is the case for the hetzner as the networking stuff doesn't have to be region specific. Moreover, hetzner doesn't allow to create the same ssh keys (they check based on the key fingerprint) so it is not possible to have more than one hetzner provider in the same cluster (unless two completely different hetzner accounts will be used in which case it will work)

**Comment by JKBGIT1** [comment]
```
alias = "nodepool_{{ $region }}_{{ $.Provider.SpecName }}"
```
And why isn't it possible to omit `{{ $region }}` in the `alias` and use only `{{ $.Provider.SpecName }}` in this line? It's for AWS, but that shouldn't matter if I got it right.

**Comment by Despire** [comment]
> And why isn't it possible to omit `{{ $region }}` in the alias and use only `{{ $.Provider.SpecName }}` in this line? It's for AWS, but that shouldn't matter if I got it right.

For some providers you need the region (for example for aws and gcp, oci..., https://github.com/berops/claudie/blob/d3c7e8c11d7af9531768a2a870667577fe1c9f4d/services/terraformer/templates/aws/provider.tpl#L5 ) as the resources are region specific. So you actually need the provider for that specific region.

**Comment by JKBGIT1** [comment]
Okay, got it. Thanks.

**Pull request merged**